mlir/lib/Gradient/IR/GradientDialect.cpp

```diff
@@ -20,6 +22,23 @@ using namespace catalyst::gradient;
 
 #include "Gradient/IR/GradientOpsDialect.cpp.inc"
 
+//===----------------------------------------------------------------------===//
+// Gradient Dialect Interfaces
+//===----------------------------------------------------------------------===//
+
+namespace {
+struct GradientInlinerInterface : public DialectInlinerInterface {
+    using DialectInlinerInterface::DialectInlinerInterface;
+
+    /// Operations in Gradient dialect are always legal to inline.
+    bool isLegalToInline(Operation *, Region *, bool,
+                         BlockAndValueMapping &valueMapping) const final
+    {
+        return true;
```

Comment:
Quick question about the inlining interface used here. If `isLegalToInline` returns true for a particular function, does this mean that that function will be inlined at all its callsites? Or will it mean that `isLegalToInline` will turn them into a candidate for inlining that still needs to undergo a profitability heuristic? If it is the second one, do you know what profitability heuristic is used?

Reply:
Cool, happy to approve.